manual/io/index.md

```diff
@@ -56,784 +46,433 @@ root [1] TFile f1("my.root");
+See also → [The `ClassDef` Macro](#the-classdef-macro)).
 
-{% highlight C++ %}
-root[] hprof->Draw()
-<TCanvas::MakeDefCanvas>: created default TCanvas with name c1
-
-root[] f->ls()
-TFile** hsimple.root
-TFile* hsimple.root
-OBJ: TProfile hprof Profile of pz versus px : 0
-KEY: TH1F hpx;1 This is the px distribution
-KEY: TH2F hpxpy;1 py vs px
-KEY: TProfile hprof;1 Profile of pz versus px
-KEY: TNtuple ntuple;1 Demo ntuple
-{% endhighlight %}
+Data members of certain types need special treatment, for instance pointers and references.
+This will be explained below.
 
-The line beginning with `OBJ` indicates that an {% include ref class="TProfile" %} object, called `hprof`, has been added in memory to this directory.
-This new `hprof` object in memory is independent from the `hprof` object on disk. If you make changes to the `hprof` object in memory, they are not propagated to the `hprof` object on disk. A new version of the `hprof` object is only saved
-if you use the `Write()` method.
+The methods of a class are not written to the ROOT file, it contains only the persistent data members.
 
-## Streamers
+See also → [Restrictions on types ROOT I/O can handle]({{ '/manual/io_custom_classes/#restrictions-on-types-root-io-can-handle' | relative_url }}).
 
-When talking about "writing an object to a file", what is actually meant is writing the current values of the data members.  The most common way to do this is to decompose (also called serialize) the object into its data elements and then write them to disk. This decomposition is done by a `Streamer`. Each class that is to be stored in a ROOT file requires a `Streamer`, which decomposes the class and "streams" its members into a buffer.
+### Generating streamers
 
-Variables of composite data types such as classes, structures, and arrays can be decomposed in simple types such as longs, shorts, floats, and chars.
+Streamers are C++ functions that are usually created as part of a class's dictionary, see → [I/O of custom classes]({{ 'manual/io_custom_classes/#generating-dictionaries' | relative_url }}).
 
-The methods of the class are not written to the ROOT file, it contains only the persistent data members. To decompose the parent classes, the `Streamer` calls the `Streamer` of the parent classes. It moves up the inheritance tree until it reaches an ancestor with no parent. To serialize the object data members, it calls their `Streamers`. These in turn move up their own inheritance tree and so on. The simple data elements are written directly to the buffer. Finally, the buffer contains all the simple data members of all the classes that make up that particular object. Data members that are references are never stored, it is always the responsibility of the object's constructor to set them correctly.
+`rootcling` parses the class definition and determines how to stream the object in an optimal way, and which streamers need to be invoked for base classes and members.
```

Comment:
First mention of `rootcling` with no links or a few words to explain. Might be that we expect readers to know/understand what this is already

Reply:
at the beginning of this document we say it "assumes that you have read the introduction on → [ROOT files]({{ '/manual/root_files/' | relative_url }}), → [I/O of custom classes]({{ '/manual/io_custom_classes/' | relative_url }}), and → [Trees]({{ '/manual/trees/' | relative_url }})".

In particular "I/O of custom classes" talks about rootcling

Reply:
I added a link to the section @jalopezg-r00t suggested, I don't think we have anything better